docs/reference/commandline/tag.md

```diff
@@ -12,36 +12,46 @@ Create a tag TARGET_IMAGE that refers to SOURCE_IMAGE
 
 ## Description
 
-An image name is made up of slash-separated name components, optionally prefixed
-by a registry hostname. The hostname must comply with standard DNS rules, but
-may not contain underscores. If a hostname is present, it may optionally be
-followed by a port number in the format `:8080`. If not present, the command
-uses Docker's public registry located at `registry-1.docker.io` by default. Name
-components may contain lowercase letters, digits and separators. A separator
-is defined as a period, one or two underscores, or one or more hyphens. A name
-component may not start or end with a separator.
-
-A tag name must be valid ASCII and may contain lowercase and uppercase letters,
-digits, underscores, periods and hyphens. A tag name may not start with a
-period or a hyphen and may contain a maximum of 128 characters.
-
-You can group your images together using names and tags, and then upload them
-to [*Share images on Docker Hub*](https://docs.docker.com/get-started/part3/).
+A full image name has the following format and components:
+
+`[REGISTRY_HOSTNAME[:REGISTRY_PORT]/]REPOSITORY[:TAG]`
+
+ - REGISTRY_HOSTNAME: The hostname must comply with standard DNS rules, but may
+not contain underscores. If the hostname is not present, the command uses
+Docker's public registry located at `registry-1.docker.io` by default.
+ - REGISTRY_PORT: If a hostname is present, it may optionally be followed by a
+port number in the format `:8080`.
+- REPOSITORY: The repository, also referred to as the image name, is made up of
+slash-separated components. Each name component may contain lowercase letters,
+digits and separators. A separator is defined as a period, one or two
+underscores, or one or more hyphens. A name component may not start or end with
+a separator. When using Docker's public registry, the repository format is
+`[USER_OR_ORGANIZATION/]REPOSITORY_NAME`. The USER_OR_ORGANIZATION is not used
+for Docker Official Images.
+ - TAG: A tag is a way to identify a specific version or variant of an image,
+such as a particular release or build. A tag name must be valid ASCII and may
+contain lowercase and uppercase letters, digits, underscores, periods and
+hyphens. A tag name may not start with a period or a hyphen and may contain a
+maximum of 128 characters.
```

Comment:
Thanks @thaJeztah, those resources are gold.
I applied your feedback and tried to update the terminology to align better with the resources you provided and Hub API spec.
I removed some of the terms from the examples because I realized I was applying Hub terminology to a local image.